src/app.ts

```diff
@@ -42,8 +42,7 @@ app.use(serviceAvailabilityMiddleware);
 app.use(`${urls.CONFIRMATION_STATEMENT}*`, sessionMiddleware);
 const userAuthRegex = new RegExp("^" + urls.CONFIRMATION_STATEMENT + "/.+");
 app.use(userAuthRegex, authenticationMiddleware);
-const companyAuthRegex = new RegExp("^" + urls.CONFIRMATION_STATEMENT + "/company/.*/");
-app.use(companyAuthRegex, companyAuthenticationMiddleware);
+app.use(`${urls.CONFIRMATION_STATEMENT}/company/:companyNumber`, companyAuthenticationMiddleware);
```

Comment:
might be worth adding a wildcard to the end of the matcher. So it matches any url starting with this

Reply:
it appears to match anything starting with that string from manual testing in the browser and i changed the unit test to check for it 
const URL = CONFIRMATION_STATEMENT + COMPANY_AUTH_PROTECTED_BASE.replace(":companyNumber", "12345678") + "/test";
That seems to work ok. I'll have to find some documentation but it looks like the string you give app.use is a starts with.
found some - https://expressjs.com/en/4x/api.html#app.use